cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -323,6 +339,61 @@ fn insert_atomic_cas(
     Ok(())
 }
 
+fn insert_shuffle(
+    fgen: &mut FunctionGenerator,
+    builder: &mut FunctionBuilder,
+    _: Opcode,
+    _: &'static [Type],
+    rets: &'static [Type],
+) -> Result<()> {
+    let ctrl_type = *rets.first().unwrap();
+
+    let lhs = builder.use_var(fgen.get_variable_of_type(ctrl_type)?);
+    let rhs = builder.use_var(fgen.get_variable_of_type(ctrl_type)?);
+
+    let mask = {
+        let lanes = fgen.u.arbitrary::<[u8; 16]>()?;
+        let lanes = ConstantData::from(lanes.as_ref());
+        builder.func.dfg.immediates.push(lanes)
+    };
+
+    let res = builder.ins().shuffle(lhs, rhs, mask);
```

Comment:
I'm tempted to assert here that the provided opcode is `Opcode::Shuffle`, since this function will be called for any opcode that uses the `Shuffle` format. I mean, yes, there's only one such opcode today, but just in case...